stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/ZipCall.kt

```diff
@@ -2,88 +2,70 @@ package io.getstream.chat.android.client.call
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
 
 internal object ZipCall {
 
-    fun <A : Any, B : Any, C : Any> zip(
-        callA: Call<A>,
-        callB: Call<B>,
-        callC: Call<C>
-    ): Call<Triple<A, B, C>> {
-        return object : ChatCallImpl<Triple<A, B, C>>() {
-            override fun execute(): Result<Triple<A, B, C>> {
-                val resultA = callA.execute()
-
-                if (!resultA.isSuccess) return Result(null, resultA.error())
-                val resultB = callB.execute()
-                if (!resultB.isSuccess) return Result(null, resultB.error())
-                val resultC = callC.execute()
-                if (!resultC.isSuccess) return Result(null, resultC.error())
-
-                return Result(Triple(resultA.data(), resultB.data(), resultC.data()), null)
-            }
+    fun <A : Any, B : Any> zip(callA: Call<A>, callB: Call<B>): Call<Pair<A, B>> {
+        return object : Call<Pair<A, B>> {
+            private var job: Job? = null
 
-            override fun enqueue(callback: (Result<Triple<A, B, C>>) -> Unit) {
-                callA.enqueue { resultA ->
-
-                    if (!resultA.isSuccess) {
-                        callback(Result(null, resultA.error()))
-                    } else {
-                        callB.enqueue { resultB ->
-                            if (!resultB.isSuccess) {
-                                callback(Result(null, resultB.error()))
-                            } else {
-
-                                callC.enqueue { resultC ->
-                                    if (!resultC.isSuccess) {
-                                        callback(Result(null, resultC.error()))
-                                    } else {
-
-                                        val dataA = resultA.data()
-                                        val dataB = resultB.data()
-                                        val dataC = resultC.data()
-
-                                        callback(Result(Triple(dataA, dataB, dataC), null))
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
+            override fun cancel() {
+                job?.cancel()
             }
-        }
-    }
 
-    fun <A : Any, B : Any> zip(callA: Call<A>, callB: Call<B>): Call<Pair<A, B>> {
-        return object : ChatCallImpl<Pair<A, B>>() {
             override fun execute(): Result<Pair<A, B>> {
-                val resultA = callA.execute()
+                val job = Job()
+                this.job = job
+
+                return runBlocking(job) {
+                    val deferredA = async { callA.await() }
+                    val deferredB = async { callB.await() }
```

Comment:
These new implementation for `ZipCall` run the two `Call`s being zipped in parallel. We only use this in a single place, and this behaviour is appropriate there.